lib/bundle-validator.js

```diff
@@ -106,6 +107,119 @@ function validateThemeConfiguration(callback) {
     callback(null, true);
 }
 
+/**
+ * Find translatable strings
+ * @param {object} schema
+ * @param {function} callback
+ */
+function getTranslatableStrings(schema, callback) {
+    var trackedKeys = ['name', 'content', 'label', 'settings', 'options'];
+
+    schema.forEach(element => {
+        Object.keys(element).forEach(key => {
+            const value = element[key];
+
+            if (trackedKeys.indexOf(key) === -1) {
+                return;
+            }
+
+            if (Array.isArray(value)) {
+                return getTranslatableStrings(value, callback);
+            }
+
+            callback(value);
+        });
+    });
+}
+
+/**
+ * Find i18n keys in schema
+ * @param {object} schema
+ * @return {array}
+ */
+function getI18nKeys(schema) {
+    var keys = [];
+
+    getTranslatableStrings(schema, (value) => {
+        if (
+            value
+            && keys.indexOf(value) === -1
+            && /^i18n\./.test(value)
+        ) {
+            keys.push(value);
+        }
+    });
+
+    return keys;
+}
+
+/**
+ * Find missed i18n keys in schemaTranslations
+ * @param {array} keys
+ * @param {object} translations
+ * @return {array}
+ */
+function findMissedKeys(keys, translations) {
+    var translationsKeys = Object.keys(translations);
+    var missingKeys = [];
+
+    keys.forEach(key => {
+        if (translationsKeys.indexOf(key) === -1) {
+            missingKeys.push(key);
+        }
+    });
+
+    return missingKeys;
+}
+
+/**
+ * Ensure that schema translations exists and there are no missing keys.
+ * @param {function} callback
+ * @return {function} callback
+ */
+function validateSchemaTranslations(callback) {
+    var v = new Validator();
+    var validatorTranslations = './schemaTranslations.schema.json';
+    var translations = {};
+    var keys = [];
+    var missedKeys;
+    var validation;
+    var errorMessage;
+
+    if (this.themeConfig.schemaExists()) {
+        keys = getI18nKeys(this.themeConfig.getRawSchema());
+    }
+
+    if (this.themeConfig.schemaTranslationsExists()) {
+        translations = this.themeConfig.getRawSchemaTranslations();
+    }
+
+    if (keys.length && _.isEmpty(translations)) {
+        errorMessage = 'Missed or corrupted schemaTranslations.json file'.red;
+
+        return callback(new Error(errorMessage));
+    }
+
+    missedKeys = findMissedKeys(keys, translations);
+    validation = v.validate(translations, require(validatorTranslations));
+
+    if ((validation.errors && validation.errors.length > 0) || (missedKeys && missedKeys.length > 0)) {
+        errorMessage = 'Your theme\'s schemaTranslations.json has errors:'.red;
+
+        missedKeys.forEach(key => {
+            errorMessage += '\r\nmissing translation key "'.red + key.red + '"'.red;
+        });
+
+        validation.errors.forEach(error => {
+            errorMessage += '\r\nschemaTranslations'.red + error.stack.substring(8).red;
```

Comment:
What is the substring call needed for?

Reply:
`error.stack` is a string with some text like this: `instance is not of a type(s) object`.
Method substring returns ` is not of a type(s) object`, so we can make an error message like this: `schemaTranslations is not of a type(s) object`